src/helpers/assertStringsWithinText.ts

```diff
@@ -0,0 +1,28 @@
+import React, { ReactElement } from 'react';
+import { ReactTestInstance } from 'react-test-renderer';
+import { filterNodeByType } from './filterNodeByType';
+
+export const assertStringsWithinText = (instance: ReactTestInstance): void => {
+  const nodesWithStringChild = instance.findAll(hasNodeStringChild);
+  nodesWithStringChild.forEach((node) => {
+    const isTextComponent = filterNodeByType(node, 'Text');
```

Comment:
This works but I get a typescript error saying this will always be false as types ElementType and string do not overlap but the typing must be incorrect since the type is a string for native component. It works when I use filterNodeByType because the node is either a ReactTestInstance or a ReactComponent, I could make the modification and add a ts-expect-error but I think it's better to use filterNodeByType and not have to explain why there is a ts error

Reply:
```suggestion
    const isHostTextComponent = filterNodeByType(node, 'Text');
```
And remove `isNativeComponent`